clients/githubrepo/client.go

```diff
@@ -192,7 +192,14 @@ func (client *Client) ListSuccessfulWorkflowRuns(filename string) ([]clients.Wor
 
 // ListCheckRunsForRef implements RepoClient.ListCheckRunsForRef.
 func (client *Client) ListCheckRunsForRef(ref string) ([]clients.CheckRun, error) {
-	return client.checkruns.listCheckRunsForRef(ref)
+	if crs, ok := client.graphClient.checkRuns[ref]; ok {
+		return crs, nil
+	}
+	crs, err := client.checkruns.listCheckRunsForRef(ref)
+	if err == nil {
+		client.graphClient.checkRuns[ref] = crs
+	}
+	return crs, err
```

Comment:
Any suggestion on how to log it?
I had tried something like this, using the scorecard log package, and it seems like it would log things for everyone, not just the cron job
```
msg := fmt.Sprintf("listCheckRunsForRef cache miss: %s/%s:%s", handler.repourl.owner, handler.repourl.repo, ref)
logger.Info(msg)
```

Reply:
I'm not too familiar with the logger package in Scorecard. Seems like it should [support](https://github.com/ossf/scorecard/blob/main/log/log.go) `logger.Debug` option? Could we use that instead?

Reply:
Staying with `logger.Info` as there is no `logger.Debug` available